tests/extra_source_and_tests/extra_src/shared/pressure_boundary/density_correciton.h

```diff
@@ -0,0 +1,119 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	fluid_surface_inner.h
+ * @brief 	Here, we define the algorithm classes for fluid surfaces.
+ * @details Fluid indicators are mainly used here to classify different region in a fluid.
+ * @author	Chi Zhang and Xiangyu Hu
+ */
+
+#ifndef DENSITY_CORRECTION_H
+#define DENSITY_CORRECTION_H
+
+#include "base_fluid_dynamics.h"
+
+namespace SPH
+{
+namespace fluid_dynamics
+{
+template <typename... InteractionTypes>
+class DensitySummationPressure;
+
+template <class DataDelegationType>
+class DensitySummationPressure<Base, DataDelegationType>
```

Comment:
the class name "DensitySummationPressure" is not easy to understand. Would do you please rethink the class name herein?

Reply:
i know this name is not exactly correct, but i did not find a good and brief name. So i hope to upload firstly, then revise it after the discussion with Dr HU